### PR TITLE
Colored Light Sprites + Colored Glowsticks + Change C-Saber Colour

### PR DESCRIPTION
**Pull request by binarysudoku**
[ENHANCEMENT]

## About the PR 

- Colored glowsticks
- QM can now order glowsticks
- Colored lights now have sprites
- You can now modify the color of a C-Saber by unscrewing and swapping out the glowstick inside for a different colour

## Why's this needed? 

- Colored light sprites are more legible and nicer to look at
- Glowsticks are fun
- Swapping C-Saber colours is fun
- The idea that C-Sabers contain actual legitimate glowsticks is fucking hilarious to me

## Changelog

```
(u)Erinexx
(*)Glowsticks are now colored. You can buy their colored variants from QM. You can also change csaber blade colors by unscrewing them and swapping out the glowstick inside!
```

### CODE REVIEW

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -889,6 +889,8 @@
 // can be tube or bulb subtypes
 // will fit into empty /obj/machinery/light of the corresponding type
 
+// TODO: SET UP ICON STATES PROPERLY!
```
Comment (ZeWaka) [inline code/modules/power/lighting.dm:892]: todo?

File: code/modules/power/lighting.dm [hunk 2]
```diff
@@ -922,104 +924,136 @@
...
+		base_state = "tube-purple"
 		color_r = 0.95
 		color_g = 0.2
 		color_b = 0.95
 	blacklight
 		name = "black light tube"
 		desc = "Fancy."
-		icon_state = "btube" // this isn't working for some reason
-		base_state = "btube"
+		icon_state = "tube-uv" // this isn't working for some reason
```
Comment (ZeWaka) [inline code/modules/power/lighting.dm:975]: is this comment relevant?

File: code/obj/item/device/lights.dm [hunk 3]
```diff
@@ -113,13 +114,19 @@
...
 	proc/burst()
 		var/turf/T = get_turf(src.loc)
 		make_cleanable( /obj/decal/cleanable/generic,T)
 		make_cleanable( /obj/decal/cleanable/greenglow,T)
 		qdel(src)
 
+	proc/turnon()
+		on = 1
+		icon_state = "[base_state][on]"
+		light.enable()
```
Comment (ZeWaka) [inline code/obj/item/device/lights.dm:128]: should be simple lights, but out of scope - would require refactor of path

File: code/obj/item/misc_weapons.dm [hunk 4]
```diff
@@ -204,25 +244,101 @@
...
+	use_glowstick = 0
 
 /obj/item/sword/discount
 	name = "d-saber"
 	desc = "A discount cyalume saber. Commonly called a d-saber."
 	state_name = "d_sword"
 	icon_state = "d_sword0"
 	item_state = "d_sword0"
-	valid_colors = list("R")
+	// valid_colors = list("R") // why was this even here
```
Comment (ZeWaka) [inline code/obj/item/misc_weapons.dm:335]: fair to just delete it then

File: code/modules/power/lighting.dm (hunk 1 above)
Comment (binarysudoku) [reply code/modules/power/lighting.dm:892]: woops
that todo is dodone

File: code/modules/power/lighting.dm (hunk 2 above)
Comment (binarysudoku) [reply code/modules/power/lighting.dm:975]: i didn't touch it but i will remove :)

File: code/obj/item/device/lights.dm (hunk 3 above)
Comment (binarysudoku) [reply code/obj/item/device/lights.dm:128]: yeah, itll be worth dedicating some time to do that in a separate PR

**Pull request merged**